cy43439.go

```diff
@@ -240,8 +242,87 @@ func (d *Device) wifiLinkStatus() int32 {
 	return whd.CYW43_LINK_DOWN
 }
 
+func (d *Device) handleAsyncEvent(payload []byte) error {
+	as, err := whd.ParseAsyncEvent(payload)
+	if err != nil {
+		return err
+	}
+	return d.processAsyncEvent(as)
+}
+
+// ref: void cyw43_ll_process_packets(cyw43_ll_t *self_in)
+func (d *Device) processPackets() error {
+	println("PROCESS PACKETS")
+	for {
+		payloadOffset, plen, header, err := d.sdpcmPoll(d.buf[:])
+		Debug("processPackets:sdpcmPoll conclude payloadoffset=", int(payloadOffset), "plen=", int(plen), "header=", header.String(), err)
+		if err != nil {
+			return err
+		}
+		payload := d.buf[payloadOffset : payloadOffset+plen]
+		switch header {
+		case whd.CONTROL_HEADER:
+			d.ctrlCh <- payload
+		case whd.ASYNCEVENT_HEADER:
+			if err := d.handleAsyncEvent(payload); err != nil {
+				return err
+			}
+		case whd.DATA_HEADER:
+			// TODO(soypat): Implement ethernet interface. cyw43_cb_process_ethernet
+			Debug("DATA_HEADER not implemented yet")
+		default:
+			Debug("doIoctl got unexpected packet", header)
+		}
+	}
+	return nil
+}
+
+// ref: bool cyw43_ll_has_work(cyw43_ll_t *self_in)
+func (d *Device) hasWork() bool {
+	return d.wlRegOn.Get()
```

Comment:
Hmmmmm- isn't the interrupt supposed to be on the [irq pin instead](https://datasheets.raspberrypi.com/picow/pico-w-datasheet.pdf?_gl=1*1es9a1*_ga*MTY0NjA5MjgyMi4xNjg3NTU4Nzk4*_ga_22FD70LWDS*MTY4NzU1ODc5Ny4xLjAuMTY4NzU1ODc5Ny4wLjAuMA..)?
![image](https://github.com/soypat/cyw43439/assets/26156425/ce347a0b-1cd7-4e1b-b8d0-f0010a008841)

Reply:
I'm not sure.  Something is not right since the driver seems to think there is always work.  I started here:

```
// Is there work to do?
bool cyw43_ll_has_work(cyw43_ll_t *self_in) {
    cyw43_int_t *self = CYW_INT_FROM_LL(self_in);
    int int_pin = cyw43_cb_read_host_interrupt_pin(self->cb_data);
    #if CYW43_USE_SPI
    return int_pin == 1;
    #else
    return int_pin == 0;
    #endif
}

```

Which calls:

```
int cyw43_cb_read_host_interrupt_pin(void *cb_data) {
    (void)cb_data;
    #ifdef CYW43_PIN_WL_HOST_WAKE
    return cyw43_hal_pin_read(CYW43_PIN_WL_HOST_WAKE);
    #else
    return cyw43_hal_pin_read(CYW43_PIN_WL_SDIO_1);
    #endif
}
```

So which pin?

Reply:
So the pico-sdk over in `pico_w.h` defines this macro as gpio24. We should use the same one. If you check the `Device` struct it already has an `irq` field for the interrupt pin which should have been set up during initialization:
![image](https://github.com/soypat/cyw43439/assets/26156425/77f04652-2eec-43f3-a3d3-6f02af9fe780)